cli/tests/activate.bats

```diff
@@ -889,3 +889,23 @@ EOF
 }
 
 # ---------------------------------------------------------------------------- #
+
+# bats test_tags=activate,activate:infinite_source,activate:infinite_source:bash
+@test "bash: test for infinite source loop" {
+  "$FLOX_BIN" delete -f
+  "$FLOX_BIN" init
+  echo 'eval "$('"$FLOX_BIN"' activate -d '"$PWD"')"' >> "$HOME/.bashrc"
+  run timeout 1 bash -c 'eval "$("$FLOX_BIN" activate)"'
```

Comment:
I feel timeout based tests can easily flake. I know building an empty env ought not take too long but if there's another way to test this i think that would be preferable.

maybe put this into the .bashrc:

```
[ $ALREADY_SOURCED == "1" ] && exit 2
export ALREADY_SOURCED=1

eval "$('"$FLOX_BIN"' activate -d '"$PWD"')"
```

which would error, if bashrc would be sourced repeatedly?

Reply:
Agreed that we should strongly prefer to avoid timeout based tests, we've worked pretty hard to excise a lot of flaky tests that were timeout based

Reply:
Feel free to revise! I'm doing eldercare things today ...